src/sql/workbench/services/connection/browser/cmsConnectionWidget.ts

```diff
@@ -97,11 +97,8 @@ export class CmsConnectionWidget extends ConnectionWidget {
 		let newAuthTypes = authTypeOption.categoryValues;
 		if (authTypeChanged) {
```

Comment:
Is this ever going to be true? I looked around the code and don't see this being called with the parameter set as true anywhere. I'm not even clear what authTypeChanged is even supposed to represent exactly...

Reply:
I'll add comments to make it clear

Reply:
I'm still not understanding what authTypeChanged is supposed to represent. Is it just for CMS? Or is it meant for general use? The base connectionWidget doesn't use it but what about other auth types in the future - will they be using this value and what is it supposed to mean? This really needs to be documented in a more central location - at the very least.

And really I'm not a fan of how this is used - as a parameter to a bunch of different methods. Shouldn't this just be a single parameter to the constructor (or even better an option in the options map) and then the class itself can use that as it needs for the various calls?